boards/FSM/Test/Src/Test_StateMachine.cpp

```diff
@@ -110,3 +128,26 @@ TEST_F(FsmStateMachineTest, check_air_closed_state_is_broadcasted_over_can)
         CANMSGS_FSM_STATE_MACHINE_STATE_AIR_CLOSED_CHOICE,
         App_CanTx_GetPeriodicSignal_STATE(can_tx_interface));
 }
+
+TEST_F(
+    FsmStateMachineTest,
+    check_if_primary_and_secondary_flow_rates_are_broadcasted_over_can_in_all_states)
+{
+    float fake_frequency = 1.0f;
+
+    for (const auto &state : GetAllStates())
+    {
+        SetInitialState(state);
+        get_flow_rate_fake.return_val = fake_frequency;
+        App_SharedStateMachine_Tick(state_machine);
+
+        EXPECT_EQ(
+            fake_frequency,
+            App_CanTx_GetPeriodicSignal_PRIMARY_FLOW_RATE(can_tx_interface));
+        EXPECT_EQ(
+            fake_frequency,
+            App_CanTx_GetPeriodicSignal_SECONDARY_FLOW_RATE(can_tx_interface));
+
+        fake_frequency++;
```

Comment:
nit; I'd leave a comment on why a `++` is needed

Reply:
Done

Reply:
done

Reply:
Wait actually now i'm confused about the comment. Isn't the reason for `++` to give a different frequency every time to tick the state machine?

Reply:
I thought of it as you can't know if the flow rate is getting updated correctly when you enter a new state if you don't change the frequency

Reply:
Right but this doesn't address WHY incrementing prevents false positive, it just claims that it DOES. It could've been incrementing, decrementing, `+=2` instead of `++`, etc but the point is you're using a different flow rate in between state machine ticks

Reply:
I updated the comment. It's the same as the comment for the `check_imd_duty_cycle_is_broadcasted_over_can_in_all_states` in IMD test.